.templates/nodered/service.yml

```diff
@@ -9,3 +9,9 @@
       - 1880:1880
     volumes:
       - ./volumes/nodered/data:/data
+      - /var/run/docker.sock:/var/run/docker.sock
+      - /var/run/dbus/system_bus_socket:/var/run/dbus/system_bus_socket
+    devices:
+      - "/dev/ttyAMA0:/dev/ttyAMA0"
+      - "/dev/vcio:/dev/vcio"
+      - "/dev/gpiomem:/dev/gpiomem"
```

Comment:
We may need to add checks to see if they exist before mapping them. I think compose will error on startup if they do not exist.

Reply:
Similar answer to the above: the paths exist on RPi3B+ and RPi4 and didn't occur to me to think about any other platforms.

That said, the magic incantation I came across when I was working through this topic also included:

```
     - "/dev/spidev0.0:/dev/spidev0.0"
     - "/dev/i2c-1:/dev/i2c-1"
     - "/dev/ttyS0:/dev/ttyS0"
```

Initially I put those three lines in place as well as the ones that are now in the PR. Nothing objected. It was only later that I worked through the list and took out the ones that didn't map to anything.

As a double-check (in case my brain was on the fritz) I have just put those lines back in place. docker-compose didn't grizzle and nodered came up OK. Not so much as a peep in nodered's log.

So, bottom line, the evidence suggests that device mappings where the left hand side doesn't exist don't cause problems. My guess is that docker is "helping" us by checking the left hand side then, if and only if the left hand side exists does it do anything with the right hand side. Sound reasonable?

Reply:
Yep, you are correct! It seems it only errors if something in `volumes` doesn't exist.